### ISSUE TITLE
Problems with M2 Macbook installation

### ISSUE THREAD

**Issue by lucas-ubm** [mlfpm/deepof#44]
I have been trying to install deepof  in my M2 MacBook air running Sonoma 14.5 and have been unable so far. 

I have followed the installation instructions (installing hdf5 and blosc and adding them to the environment) but I still get the following error when doing poetry install in a conda environment with python=3.9

`  
  tables/utilsextension.pyx:384:47: Cannot assign type 'herr_t (unsigned int, const H5E_error_t *, void *) except? -1 nogil' to 'H5E_walk_t' (alias of 'herr_t (*)(unsigned int, H5E_error_t *, void *) noexcept nogil'). Exception values are incompatible. Suggest adding 'noexcept' to the type of 'e_walk_cb'.
  cpuinfo failed, assuming no CPU features: 'flags'
  * Using Python 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:55:20) 
  * Found cython 3.0.10
  * USE_PKGCONFIG: True
  * Found conda env: ``/Users/admin/miniconda3/envs/deepof``
  * pkg-config header dirs for HDF5: /opt/homebrew/Cellar/hdf5/1.14.3_1/include
  * pkg-config library dirs for HDF5: /opt/homebrew/Cellar/hdf5/1.14.3_1/lib
  * Found HDF5 headers at ``/opt/homebrew/Cellar/hdf5/1.14.3_1/include``, library at ``/opt/homebrew/Cellar/hdf5/1.14.3_1/lib``.
  * Found LZO 2 headers at ``/Users/admin/miniconda3/envs/deepof/include``, library at ``/Users/admin/miniconda3/envs/deepof/lib``.
  * Skipping detection of LZO 1 since LZO 2 has already been found.
  * Found bzip2 headers at ``/Users/admin/miniconda3/envs/deepof/include``, library at ``/Users/admin/miniconda3/envs/deepof/lib``.
  * pkg-config header dirs for blosc: /opt/homebrew/Cellar/c-blosc/1.21.5/include
  * pkg-config library dirs for blosc: /opt/homebrew/Cellar/c-blosc/1.21.5/lib
  * Found blosc headers at ``/opt/homebrew/Cellar/c-blosc/1.21.5/include``, library at ``/opt/homebrew/Cellar/c-blosc/1.21.5/lib``.
  * Found blosc2 headers at ``/var/folders/7w/lttqd6qs2cz9l_264jz_tb600000gn/T/tmpyjdojxj2/.venv/include``, library at ``/var/folders/7w/lttqd6qs2cz9l_264jz_tb600000gn/T/tmpyjdojxj2/.venv/lib``.
    * Copying blosc2 runtime library to 'tables' dir because it was not found in standard locations
  Compiling tables/utilsextension.pyx because it changed.
  [1/1] Cythonizing tables/utilsextension.pyx
  Traceback (most recent call last):
    File "/Users/admin/.local/pipx/venvs/poetry/lib/python3.9/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in <module>
      main()
    File "/Users/admin/.local/pipx/venvs/poetry/lib/python3.9/site-packages/pyproject_hooks/_in_process/_in_process.py", line 357, in main
      json_out["return_val"] = hook(**hook_input["kwargs"])
    File "/Users/admin/.local/pipx/venvs/poetry/lib/python3.9/site-packages/pyproject_hooks/_in_process/_in_process.py", line 134, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/var/folders/7w/lttqd6qs2cz9l_264jz_tb600000gn/T/tmpyjdojxj2/.venv/lib/python3.9/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
    File "/var/folders/7w/lttqd6qs2cz9l_264jz_tb600000gn/T/tmpyjdojxj2/.venv/lib/python3.9/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
      self.run_setup()
    File "/var/folders/7w/lttqd6qs2cz9l_264jz_tb600000gn/T/tmpyjdojxj2/.venv/lib/python3.9/site-packages/setuptools/build_meta.py", line 311, in run_setup
      exec(code, locals())
    File "<string>", line 928, in <module>
    File "<string>", line 923, in get_cython_extfiles
    File "/var/folders/7w/lttqd6qs2cz9l_264jz_tb600000gn/T/tmpyjdojxj2/.venv/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1154, in cythonize
      cythonize_one(*args)
    File "/var/folders/7w/lttqd6qs2cz9l_264jz_tb600000gn/T/tmpyjdojxj2/.venv/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1321, in cythonize_one
      raise CompileError(None, pyx_file)
  Cython.Compiler.Errors.CompileError: tables/utilsextension.pyx
`

I have also tried to run things in Docker but since a GUI is needed for determining the area coordinates I have not been able to complete the starting of the project.

**Comment by lucas-ubm** [comment]
Dear Lucas,

Thanks for the speedy response! For some extra context, I have attempted (without success): 

- pip installation
- conda installation of hdf5 and blosc 
- independent installation of tables library from pip

**Comment by lucasmiranda42** [comment]
Thank you! I just managed to install the last version of the package from PyPI with these commands:

```bash
conda create -n deepof python=3.10
conda activate deepof
conda install -c anaconda pytables==3.8.0
pip install deepof
```

The key seems to be installing the exact same version of pytables with conda beforehand. Feel free to give it a try! We'll test it a bit further and add it to the documentation as soon as we're sure it works.

Best, hope it helps, and please let us know if it does not!
Lucas

**Comment by lucasmiranda42** [comment]
Hi, @lucas-ubm!

Hope you're doing well. Did you manage to test the fix by any chance? Your feedback would be highly appreciated :)

Best, and thank you!
Lucas

**Comment by lucas-ubm** [comment]
Hi @lucasmiranda42 ,

Thank you for the fix and sorry for the late response. The installation succeeded but I get an error when importing deepof.visuals. I'm not sure if it's still related with the installation or something else completely. 

Here is the traceback:
`Traceback (most recent call last):
  File "test.py", line 1, in <module>
    import deepof.visuals
  File "/Users/admin/miniconda3/envs/deepof/lib/python3.10/site-packages/deepof/visuals.py", line 33, in <module>
    import deepof.post_hoc
  File "/Users/admin/miniconda3/envs/deepof/lib/python3.10/site-packages/deepof/post_hoc.py", line 29, in <module>
    import ot
  File "/Users/admin/miniconda3/envs/deepof/lib/python3.10/site-packages/ot/__init__.py", line 24, in <module>
    from . import optim
  File "/Users/admin/miniconda3/envs/deepof/lib/python3.10/site-packages/ot/optim.py", line 23, in <module>
    from scipy.optimize.linesearch import scalar_search_armijo
ImportError: cannot import name 'scalar_search_armijo' from 'scipy.optimize.linesearch' (/Users/admin/miniconda3/envs/deepof/lib/python3.10/site-packages/scipy/optimize/linesearch.py)
`

**Comment by lucasmiranda42** [comment]
Dear @lucas-ubm,

My apologies for the delay. We confirmed the error you posted above, and fixed it in the new 0.6.5 release. The latest version should be installable both via poetry and pip. The documentation has also been updated :)

Best, and feel free to reopen if necessary!
Lucas
